File: backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/appeal/Appeal.java
package de.bund.digitalservice.ris.caselaw.domain.appeal;

import java.util.List;
import java.util.UUID;
import lombok.Builder;

@Builder
public record Appeal(
    UUID id,
    List<Appellant> appellants,
    List<AppealStatus> revisionDefendantStatuses,
    List<AppealStatus> revisionPlaintiffStatuses,
    List<AppealStatus> jointRevisionDefendantStatuses,
    List<AppealStatus> jointRevisionPlaintiffStatuses,
    List<AppealStatus> nzbDefendantStatuses,
    List<AppealStatus> nzbPlaintiffStatuses,
    AppealWithdrawal appealWithdrawal,
    PkhPlaintiff pkhPlaintiff) {}

File: frontend/src/domain/appeal.ts
import { DropdownItem } from "@/components/input/types"

export type Appeal = {
  id?: string
  appellants?: Appellant[]
  revisionDefendantStatuses?: AppealStatus[]
  revisionPlaintiffStatuses?: AppealStatus[]
  jointRevisionDefendantStatuses?: AppealStatus[]
  jointRevisionPlaintiffStatuses?: AppealStatus[]
  nzbDefendantStatuses?: AppealStatus[]
  nzbPlaintiffStatuses?: AppealStatus[]
  appealWithdrawal?: AppealWithdrawal
  pkhPlaintiff?: PkhPlaintiff
}

export type Appellant = {
  id: string
  value: string
}

export type AppealStatus = {
  id: string
  value: string
}

export enum AppealWithdrawal {
  JA = "JA",
  NEIN = "NEIN",
  KEINE_ANGABE = "KEINE_ANGABE",
}
export const appealWithdrawalItems: DropdownItem[] = [
  {
    label: "Ja",
    value: AppealWithdrawal.JA,
  },
  {
    label: "Nein",
    value: AppealWithdrawal.NEIN,
  },
  {
    label: "Keine Angabe",
    value: AppealWithdrawal.KEINE_ANGABE,
  },
]
export enum PkhPlaintiff {
  JA = "JA",
  NEIN = "NEIN",
  KEINE_ANGABE = "KEINE_ANGABE",
}
export const pkhPlaintiffItems: DropdownItem[] = [
  {
    label: "Ja",
    value: PkhPlaintiff.JA,
  },
  {
    label: "Nein",
    value: PkhPlaintiff.NEIN,
  },
  {
    label: "Keine Angabe",
    value: PkhPlaintiff.KEINE_ANGABE,
  },
]
